staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -279,6 +279,10 @@ type childNode struct {
 	jsonName       string
 	underlyingType *types.Type
 	validations    []validators.FunctionGen
+
+	// eachVal and eachKey validation is tracked here since attaching it directly to the
```

Comment:
Ouch.  I don't love this.  It shouldn't be sharing, I want to look at at that.

Reply:
I dug into this and I realized the flub.   I ended up creating the same basic approach as you.  I want to tweak on it a little more, but yeah, you had the right of it.  I still don't love it, though